src/lib/pages/migrate/components/MigrateContract.tsx

```diff
@@ -185,6 +182,13 @@ export const MigrateContract = observer(
       return () => {};
     }, [address, codeId, contractAddress, enableMigrate, currentInput]);
 
+    useEffect(() => {
+      if (!isCodeId(codeId)) {
+        setValue("codeHash", "");
+        setTab(MessageTabs.JSON_INPUT);
+      }
```

Comment:
1. Could we merge these functionalities with the useEffect on lines 155-166?
2. Using `isCodeId` for validation isn't effective. If a user modifies their input to another valid code id, it doesn't reset as expected.
3. Upon debuging, I've noticed the debounce mechanism isn't functioning as intended.

Reply:
2. so the validation is not needed at all? remove?
3. please explain more in details